src/substrait/textplan/parser/SubstraitPlanRelationVisitor.cpp

```diff
@@ -2075,10 +2087,19 @@ void SubstraitPlanRelationVisitor::applyOutputMappingToSchema(
   if (common->emit().output_mapping_size() == 0) {
     common->mutable_direct();
   } else {
-    if (!relationData->outputFieldReferences.empty()) {
-      // TODO -- Add support for aggregate relations.
-      errorListener_->addError(
-          token, "Aggregate relations do not yet support emit sections.");
+    if (relationData->relation.has_aggregate()) {
```

Comment:
This looks very similar to the code at the end of `InitialPlanProtoVisitor::updateLocalSchema`. Any chance it could be extracted to a helper function?